libraries/AP_InertialSensor/AP_InertialSensor_SensorHead.cpp

```diff
@@ -0,0 +1,56 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_BoardConfig/AP_BoardConfig.h>
+
+#include "AP_InertialSensor_SensorHead.h"
+
+extern const AP_HAL::HAL& hal;
+
+void InertialSensorMessageHandler::handle(InertialSensorMessage::data_t *data)
+{
+    if (_backend->_sem_ins->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+        _backend->_gyro.x = data->gyrox;
+        _backend->_gyro.y = data->gyroy;
+        _backend->_gyro.z = data->gyroz;
+        _backend->_accel.x = data->accelx;
+        _backend->_accel.y = data->accely;
+        _backend->_accel.z = data->accelz;
+        _backend->_last_timestamp = AP_HAL::micros64();
+        _backend->_sem_ins->give();
+    }
+}
+
+AP_InertialSensor_SensorHead::AP_InertialSensor_SensorHead(AP_InertialSensor &imu) :
+    AP_InertialSensor_Backend(imu)
+{
+    // TODO: What is the correct id here? Is 0 ok?
```

Comment:
we should either pass the sensor ID over in the protocol or assign a DEVTYPE_* for SensorHead, Or perhaps 3 IDs for SensorHead, for 1st, 2nd and 3rd IMU?